### PR TITLE
Problem tutorial and web site improvements

### PR DESCRIPTION
**Pull request by darioizzo**

### CODE REVIEW

File: doc/sphinx/docs/examples/getting_started.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    archipelago archi{16, algo, prob, 20};
+
+    // 4 - Run the evolution in parallel on the 16 separate islands 10 times.
+    archi.evolve(10);
+
+    // 5 - Wait for the evolutions to be finished
+    archi.wait();
+
+    // 6 - Print the fitness of the best solution in each island
+    for (const isl & : archi) {
```
Comment (bluescarni) [inline doc/sphinx/docs/examples/getting_started.cpp]: Should we run clang-format on these examples?

File: doc/sphinx/docs/python/tutorials/coding_udp_simple.rst [hunk 2]
```diff
@@ -0,0 +1,258 @@
+.. _py_tutorial_coding_udp_simple:
+
+Coding a simple User Defined Problem
+------------------------------------
+
+While pagmo provides a number of UDPs to help you test your own optimization strategy or user defined algorithm, the possibility
+to write your own UDP is at the core of pygmo's use. In this tutorial we will show how to code a UDP. Remember that UDPs are classes that can be used 
+to construct a :class:`~pygmo.core.problem` which, in turn, is what a :class:`~pygmo.core.algorithm` can solve.
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/coding_udp_simple.rst]: "a algorithm" -> "an algorithm"

File: doc/sphinx/docs/python/tutorials/coding_udp_simple.rst [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+.. doctest::
+
+    >>> class sphere_function:
+    ...     def fitness(self, x):
+    ...         return [sum(x*x)]
+    ...         
+    ...     def get_bounds(self):
+    ...         return ([-1,-1],[1,1])
+
+The two mandatory methods you must implement in your class are ``fitness(self, x)`` and ``def get_bounds(self)``. The problem dimension
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/coding_udp_simple.rst]: We should decide if we want the ``def`` or not here, just to be consistent.

File: doc/sphinx/docs/python/tutorials/coding_udp_simple.rst [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+will be inferred by the return value of the second, while the actual fitness of decision vectors will be computed calling the first method 
+and passing as argument ``x`` as a NumPy array. It is important to remember that ``x`` is a NumPy array, so that the NumPy
+array arithmetic applies in the body of ``fitness``. Note also how to define a UDP we do not need to inherit from some other
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/coding_udp_simple.rst]: ``fitness`` -> ``fitness()``

File: doc/sphinx/docs/python/tutorials/coding_udp_simple.rst [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+pygmo related class.  Since we do not define, in this case, any other method pygmo will assume a single objective, no constraints,
+no gradients etc...
+
+Lets now build a :class:`~pygmo.core.problem` from our new UDP.
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/coding_udp_simple.rst]: "Let's"

File: doc/sphinx/docs/python/tutorials/coding_udp_simple.rst [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+    	User implemented gradient sparsity: false
+    	Has hessians: false
+    	User implemented hessians sparsity: false
+    <BLANKLINE>
+    	Function evaluations: 0
+    <BLANKLINE>
+    	Thread safety: none
+    <BLANKLINE>
+
+Lets now add some (mild) complexity. We want our UDP to be scalable:
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/coding_udp_simple.rst]: "Let's"

File: doc/sphinx/docs/python/tutorials/coding_udp_simple.rst [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+    >>> print(pop.champion_f) #doctest: +SKIP
+    [  3.75822114e-06]
+
+Wow those bees!! 
+
+Possible pitfalls
+^^^^^^^^^^^^^^^^^
+
+Well that was nice as it worked like a charm. But the UDP can also be a rather complex class and the chances
+that it is some how malformed are high. Lets see some common mistakes.
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/coding_udp_simple.rst]: "somehow", "Let's"

File: doc/sphinx/docs/python/tutorials/coding_udp_simple.rst [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+    ...         return [sum(x*x)]
+    ...         
+    >>> pg.problem(sphere_function()) #doctest: +SKIP
+    NotImplementedError                       Traceback (most recent call last)
+    ...
+    NotImplementedError: the mandatory 'get_bounds()' method has not been detected in the user-defined Python problem
+    '<sphere_function object at 0x1108cad68>' of type '<class 'sphere_function'>': the method is either not present or not callable
+
+
+oops, I forgot to implement one of the two mandatory methods. In this case its not possible to construct a :class:`~pygmo.core.problem`
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/coding_udp_simple.rst]: its -> it is

File: doc/sphinx/docs/python/tutorials/coding_udp_simple.rst [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+    ...
+    AttributeError: 'numpy.float64' object has no attribute '__iter__'
+
+Notes on computational speed
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The most performant way to write a UDP is to code it in C++ and expose it to python. Most UDPs that
+are included in pygmo (see :ref:`py_problems`) are like that. When writing your own UDP, though, it is often quicker and less
+painful to code, as shown in this tutorial, directly in python. What effect does this have w.r.t. the ideal
+situation? Well, lets see, on a test machine, a simple example: the scalable Rosenbrock function:
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/coding_udp_simple.rst]: "Let's"

File: doc/sphinx/docs/python/tutorials/coding_udp_simple.rst [hunk 10]
```diff
@@ -0,0 +1,258 @@
...
+    >>> prob_cpp = pg.problem(pg.rosenbrock(2000))
+    >>> dummy_x = np.full((2000,), 1.)
+    >>> import time
+    >>> start_time = time.time(); [prob_python.fitness(dummy_x) for i in range(1000)]; print(time.time() - start_time) #doctest: +SKIP
+    2.3352...
+    >>> start_time = time.time(); [prob_cpp.fitness(dummy_x) for i in range(1000)]; print(time.time() - start_time) #doctest: +SKIP
+    0.0114226...
+
+wait a minute ... really? Python is two orders of magnitude slower than cpp? Do not panic. This is a very large problem and that for loop is not going to be
+super optimized in python. Lets see if we can do better in these cases .... Let us use the jit decorator from numba to compile
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/coding_udp_simple.rst]: "Let's"

File: doc/sphinx/docs/python/tutorials/using_population.rst [hunk 11]
```diff
@@ -91,6 +91,15 @@ length.
 .. note:: Decision vectors that are outside of the box bounds are allowed to be
           pushed back into a population
 
+The snippet above will trigger fitness function evaluations as the decision vector is always associated to a fitness vector in a pop:func:`~pygmo.core.population`.
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/using_population.rst]: ``pygmo.core.population`` is a ``:class:``, not a ``:func:``, and it looks like a ``pop`` slipped in at the beginning.

File: doc/sphinx/docs/python/tutorials/using_population.rst [hunk 12]
```diff
@@ -111,8 +120,8 @@ some individual decision vector:
     >>> print(pop1.get_f()[0])
     [ 8.43469444]
 
-.. note:: Using the method :class:`~pygmo.core.population.set_xf` it is possible to avoid
-          triggering the fitness function evaluation, but it is also possible to inject
+.. note:: Using the method :class:`~pygmo.core.population.set_xf` or:class:`~pygmo.core.population.push_back` it is possible to avoid
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/using_population.rst]: These are methods and not classes, and the ``()`` round brackets are missing at the end.

File: doc/sphinx/install.rst [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+
+.. code-block:: bash
+
+   pip install pygmo
+
+or
+
+.. code-block:: bash
+
+   conda install pygmo
```
Comment (bluescarni) [inline doc/sphinx/install.rst]: I am not 100% sure if this will work without adding the conda-forge channel first.

File: doc/sphinx/install.rst [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+
+Building the module
+^^^^^^^^^^^^^^^^^^^
+
+To build the module you need to have the Boost Python libraries installed and to activate the BUILD_PYGMO option from within ``cmake``.
```
Comment (bluescarni) [inline doc/sphinx/install.rst]: BUILD_PYGMO should be in backticks ``BUILD_PYGMO``

File: doc/sphinx/install.rst [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+
+   git clone https://github.com/esa/pagmo2.git
+
+and configure your build using ``cmake``. When done, type (in your build directory):
+
+.. code-block:: bash
+
+   make install
+
+The headers will be installed in the CMAKE_INSTALL_PREFIX/include directory. To check that all went well compile the :ref:`quick-start example <getting_started_c++>`.
```
Comment (bluescarni) [inline doc/sphinx/install.rst]: CMAKE_INSTALL_PREFIX/include should probably be in backticks.

File: doc/sphinx/install.rst [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+
+Building the module
+^^^^^^^^^^^^^^^^^^^
+
+To build the module you need to have the Boost Python libraries installed and to activate the BUILD_PYGMO option from within ``cmake``.
+
+Check carefully what Python version is detected and what libraries are linked to. In particular select the correct boost_python
+according to the Python version (2 or 3) you want to compile the module for.
+
+The CMAKE_INSTALL_PREFIX will be used to construct the final location of headers and Python module after install.
```
Comment (bluescarni) [inline doc/sphinx/install.rst]: Ditto.

File: doc/sphinx/docs/examples/getting_started.cpp (hunk 1 above)
Comment (darioizzo) [inline doc/sphinx/docs/examples/getting_started.cpp]: Yes, but I think it did run

Comment (bluescarni) [inline doc/sphinx/docs/examples/getting_started.cpp]: Shouldn't that be ``const auto &isl`` or ``const island &isl``?

File: doc/sphinx/docs/python/tutorials/coding_udp_simple.rst [hunk 17]
```diff
@@ -0,0 +1,258 @@
...
+
+Well that was easy, but now have a :class:`~pygmo.core.problem` to solve ... 
+
+    >>> algo = pg.algorithm(pg.bee_colony(gen = 20, limit = 20))
+    >>> pop = pg.population(prob,10)
+    >>> pop = algo.evolve(pop)
+    >>> print(pop.champion_f) #doctest: +SKIP
+    [  3.75822114e-06]
+
+Wow those bees!!
```
Comment (bluescarni) [inline doc/sphinx/docs/python/tutorials/coding_udp_simple.rst]: I think this is my favourite line from the  tutorial... "Look at them go!"

File: doc/sphinx/install.rst [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+-----------------------------------------------------------------------
+
+Python
+------
+The python module correponding to pagmo is called pygmo
+It can be installed either directly from ``conda`` or ``pip`` or by building the module from source.
+
+Installing with pip
+^^^^^^^^^^^^^^^^^^^
+The Python package pygmo (Python binding of the C++ code) can be installed using ``pip`` or ``conda``:
```
Comment (bluescarni) [inline doc/sphinx/install.rst]: I think everywhere else we say ``python`` without capitals? Or maybe not...

File: doc/sphinx/docs/examples/getting_started.cpp (hunk 1 above)
Comment (darioizzo) [inline doc/sphinx/docs/examples/getting_started.cpp]: of course but. But clang format does not catch these things

**Pull request merged**